cpp/src/arrow/ipc/metadata_internal.cc

```diff
@@ -427,8 +427,7 @@ static Status GetDictionaryEncoding(FBB& fbb, const std::shared_ptr<Field>& fiel
                                     const DictionaryType& type, int64_t dictionary_id,
                                     DictionaryOffset* out) {
   // We assume that the dictionary index type (as an integer) has already been
-  // validated elsewhere, and can safely assume we are dealing with signed
-  // integers
+  // validated elsewhere, and can safely assume we are dealing with integers
```

Comment:
We probably shouldn't change this from the CPP side

Reply:
This is just changing a comment to align with what the code is actually doing. I initially read the comment and thought the Rust code should assume it's dealing only with signed integers, and then I read the CPP code and realized the comment was out of date. This should have been updated with b1a7a73. 

I'm happy to pull [this commit](https://github.com/apache/arrow/pull/8402/commits/5b271bd55a58d5adfde00ea4b6505f4466fe7d9a) out into a separate PR if you'd like?

Reply:
Yes please, unless it's fine @emkornfield ?

Reply:
I think we should remove it, touching anything in `./cpp` triggers the C++, Python and other CI, which are currently failing because of LLVM issues that have still exist in the parquet branch.